src/main.cpp

```diff
@@ -418,40 +638,50 @@ void emulator_loop()
 		//
 		// Trace functionality preserved for comparison with official emulator releases
 		//
-#if defined(TRACE)
+//#if defined(TRACE)
 		{
-			printf("[%6d] ", instruction_counter);
-
-			char *label     = label_for_address(state6502.pc);
-			int   label_len = label ? strlen(label) : 0;
-			if (label) {
-				printf("%s", label);
-			}
-			for (int i = 0; i < 20 - label_len; i++) {
-				printf(" ");
-			}
-			printf(" %02x:.,%04x ", memory_get_rom_bank(), state6502.pc);
-			char disasm_line[15];
-			int  len = disasm(state6502.pc, disasm_line, sizeof(disasm_line), 0);
-			for (int i = 0; i < len; i++) {
-				printf("%02x ", debug_read6502(state6502.pc + i));
-			}
-			for (int i = 0; i < 9 - 3 * len; i++) {
-				printf(" ");
+			uint16_t pc = state6502.pc;
+			uint8_t  x  = state6502.x;
+			uint8_t  y  = state6502.y;
+			uint8_t  a  = state6502.a;
+			uint8_t  status  = state6502.status;
+			uint8_t  sp      = state6502.sp;
+			uint8_t  ram     = memory_get_ram_bank();	
+			uint8_t  rom     = memory_get_rom_bank();	
+			uint8_t  cur     = memory_get_current_bank(pc);	
+			char     buffer[256];
+			uint8_t  pos = 0;
+
+			if ( (Options.log_cpu_main && (pc >= 0x0800 && pc <= 0x9FFF ) ) ||
+			     (Options.log_cpu_bram && (pc >= 0xA000 && pc <= 0xBFFF ) ) ||
+			     (Options.log_cpu_low  && (pc >= 0x0000 && pc <= 0x07FF ) ) ||
+ 			     (Options.log_cpu_brom && (pc >= 0xC000 && pc <= 0xFFFF ) )
+				) {
+
+				 printf("a:$%02x x:$%02x y:$%02x s:$%02x p:", a, x, y, sp);
+				 for (int i = 7; i >= 0; i--) {
+					 printf("%c", (status & (1 << i)) ? "czidb.vn"[i] : '-');
+				 }
+
+				 printf(" ram=$%02x rom=$%02x ", ram, rom);
+				 const char *label     = disasm_get_label(pc);
+				 size_t      label_len = label ? strlen(label) : 0;
+				 if (label) {
+					 printf("%s", label);
+				 }
+				 label_len = (label_len <= 20) ? label_len : 20;
+				 for (int i = 0; i < 20 - label_len; i++) {
```

Comment:
This is where the Windows build is breaking. All build warnings are errors, and on Windows the `size_t` type is a `uint64_t` while `int` is an `int32_t`, resulting in a signed/unsigned mismatch in this comparison. I'd suggest changing `i` to a `size_t` to match `label_len`.

Reply:
Thanks. Fixing.